source/common/http/http1/codec_impl.cc

```diff
@@ -878,6 +879,30 @@ int ServerConnectionImpl::onHeadersComplete() {
           "http/1.1 protocol error: request headers failed spec compliance checks");
     }
 
+    // https://tools.ietf.org/html/rfc7230#section-3.3.3
+    // If a message is received with both a Transfer-Encoding and a
+    // Content-Length header field, the Transfer-Encoding overrides the
+    // Content-Length. Such a message might indicate an attempt to
+    // perform request smuggling (Section 9.5) or response splitting
+    // (Section 9.4) and ought to be handled as an error. A sender MUST
+    // remove the received Content-Length field prior to forwarding such
+    // a message.
+
+    // Reject request with Http::Code::BadRequest by default or remove Content-Length header
+    // and serve request if allowed by http1 codec settings.
+    if (parser_.uses_transfer_encoding != 0 &&
+        (parser_.content_length > 0 && parser_.content_length != ULLONG_MAX)) {
```

Comment:
what is the parser_.content_length != ULLONG_MAX about?

Also, is this case handled correctly?
transfer-encoding: chunked
content-length: 0

Please add a test with content-length == 0.

Reply:
Updated code, added testcases with different Content-Length.